### PR TITLE
Determines if predictor is using MMS by checking inference ConfigMap

### PR DESCRIPTION
**Pull request by abchoo**
**What this PR does / why we need it**:  Previously, flag for MMS set in the ConfigMap did not determine if a predictor uses MMS

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by yuzisun** [comment]
@PatrickXYS We are still getting authorization error 401 a lot, any idea what could be the issue?

```
E                   RuntimeError: Exception when calling CustomObjectsApi->get_namespaced_custom_object:                        (401)
E                   Reason: Unauthorized
E                   HTTP response headers: HTTPHeaderDict({'Audit-Id': '3913542d-6604-43df-9f65-12e4ec8d94a7', 'Cache-Control': 'no-cache, private', 'Cont
ent-Type': 'application/json', 'Date': 'Wed, 03 Feb 2021 22:54:42 GMT', 'Content-Length': '129'})
E                   HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorize
d","code":401}
```

**Comment by PatrickXYS** [comment]
I checked the YAML file, it expose AWS credentials for sure.

I'm guessing it's a random issue, otherwise, all the test cases in https://github.com/kubeflow/kfserving/blob/master/test/scripts/run-e2e-tests.sh will fail for `Unauthorized` error

**Comment by yuzisun** [comment]
> I checked the YAML file, it expose AWS credentials for sure.
> 
> I'm guessing it's a random issue, otherwise, all the test cases in https://github.com/kubeflow/kfserving/blob/master/test/scripts/run-e2e-tests.sh will fail for `Unauthorized` error

ye it is random but we hit this quite a lot, not sure about the reason though.

**Comment by PatrickXYS** [comment]
> > I checked the YAML file, it expose AWS credentials for sure.
> > I'm guessing it's a random issue, otherwise, all the test cases in https://github.com/kubeflow/kfserving/blob/master/test/scripts/run-e2e-tests.sh will fail for `Unauthorized` error
> 
> ye it is random but we hit this quite a lot, not sure about the reason though.

I agree, I haven't seen same error in other repositories yet. Should be the issue hitted by large-volume workload from KFServing repo I assume? 

Let me check if I can help debug why it happened

**Comment by PatrickXYS** [comment]
I think most of the issue happens when 

```
    KFServing.create(isvc)
    try:
        KFServing.wait_isvc_ready(service_name, namespace=KFSERVING_TEST_NAMESPACE, 
                                version=constants.KFSERVING_VERSION)
    except RuntimeError as e:
        KFServing.wait_isvc_ready(service_name, namespace=KFSERVING_TEST_NAMESPACE, 
                                version=constants.KFSERVING_V1BETA1_VERSION)
```

https://github.com/kubeflow/kfserving/blob/master/test/e2e/predictor/test_lightgbm.py#L60-L66

**Comment by PatrickXYS** [comment]
Can we add some print stuff in `KFServing.wait_isvc_ready` function?

E.g, 

```
if os.getenv("AWS_ACCESS_KEY_ID"):
  print("AWS_ACCESS_KEY_ID has been set up")
else:
  print("AWS_ACCESS_KEY_ID is not set up yet or disappeared") 
```

I think to handle the issue, we need to first locate the issue and why it happened

**Comment by PatrickXYS** [comment]
Seems like you missed `import os`

/cc @abchoo

**Comment by PatrickXYS** [comment]
```
FAILED predictor/test_pmml.py::test_pmml_kfserving - NameError: name 'os' is ...
FAILED predictor/test_pytorch.py::test_pytorch - NameError: name 'os' is not ...
FAILED predictor/test_sklearn.py::test_sklearn_kfserving - NameError: name 'o...
FAILED predictor/test_sklearn.py::test_sklearn_v2_kfserving - NameError: name...
FAILED predictor/test_torchserve.py::test_torchserve_kfserving - NameError: n...
FAILED predictor/test_tensorflow.py::test_tensorflow_kfserving - NameError: n...
FAILED predictor/test_xgboost.py::test_xgboost_v2_kfserving - NameError: name...
FAILED predictor/test_triton.py::test_triton - NameError: name 'os' is not de...
FAILED predictor/test_xgboost.py::test_xgboost_kfserving - NameError: name 'o...
FAILED transformer/test_transformer.py::test_transformer - NameError: name 'o...
```

**Comment by abchoo** [comment]
Yeah, thanks for the heads up! I thought I made another commit to fix that but I guess I didn't.

**Comment by abchoo** [comment]
> Can we add some print stuff in `KFServing.wait_isvc_ready` function?
> 
> E.g,
> 
> ```
> if os.getenv("AWS_ACCESS_KEY_ID"):
>   print("AWS_ACCESS_KEY_ID has been set up")
> else:
>   print("AWS_ACCESS_KEY_ID is not set up yet or disappeared") 
> ```
> 
> I think to handle the issue, we need to first locate the issue and why it happened

Just to save you some time here is the error + output for the transformer test failure
```
>           raise ApiException(http_resp=r)
E           kubernetes.client.exceptions.ApiException: (401)
E           Reason: Unauthorized
E           HTTP response headers: HTTPHeaderDict({'Audit-Id': '23978c3c-8209-41ac-bc24-17a452a2b858', 'Cache-Control': 'no-cach
e, private', 'Content-Type': 'application/json', 'Date': 'Thu, 04 Feb 2021 05:28:49 GMT', 'Content-Length': '129'})
E           HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","re
ason":"Unauthorized","code":401}
/usr/local/lib/python3.8/dist-packages/kubernetes/client/rest.py:233: ApiException
------------------------------ Captured log call -------------------------------
WARNING  root:kf_serving_client.py:101 The version v1alpha2 will be deprecated from KFServing 0.6 release.
WARNING  root:kf_serving_client.py:318 AWS_ACCESS_KEY_ID has been set up
```

File: config/overlays/test/configmap/inferenceservice.yaml [hunk 1]
```diff
@@ -53,20 +59,23 @@ data:
...
             "image": "809251082950.dkr.ecr.us-west-2.amazonaws.com/kfserving/pmmlserver",
-            "defaultImageVersion": "latest"
+            "defaultImageVersion": "latest",
+            "multiModelServer": false
         },
         "lightgbm": {
             "image": "809251082950.dkr.ecr.us-west-2.amazonaws.com/kfserving/lgbserver",
-            "defaultImageVersion": "latest"
+            "defaultImageVersion": "latest",
+            "multiModelServer": false
```
[the remark this answers is not in the document]
Comment (abchoo) [reply config/overlays/test/configmap/inferenceservice.yaml]: Good catch

**Comment by PatrickXYS** [comment]
I saw one of the test round succeeded 9 hours ago 

https://argo.kubeflow-testing.com/workflows/kubeflow-test-infra/kubeflow-kfserving-presubmit-e2e-1339-406eabf-7344-6f28

Let's see how it goes with the latest one

**Comment by PatrickXYS** [comment]
How about setting retry mechanism, every time we see `Unauthorized 401`  exception, we send another API call. 

Or even simpler for experimenting

E.g, 

```
    KFServing.create(isvc)
    try:
        KFServing.wait_isvc_ready(service_name, namespace=KFSERVING_TEST_NAMESPACE, 
                                version=constants.KFSERVING_VERSION)
    except:
        KFServing.wait_isvc_ready(service_name, namespace=KFSERVING_TEST_NAMESPACE, 
                                version=constants.KFSERVING_V1BETA1_VERSION)
```

**Comment by PatrickXYS** [comment]
@abchoo We can move forward on this PR

If you're insterested in adding retry, you can start on a new PR instead

File: config/configmap/inferenceservice.yaml [hunk 2]
```diff
@@ -31,15 +31,15 @@ data:
             "supportedFrameworks": [
               "sklearn"
             ],
-            "multiModelServer": "true"
+            "multiModelServer": true
```
Comment (yuzisun) [inline config/configmap/inferenceservice.yaml:34]: @yuzliu should we default these to false when making the release and have people turn on manually if they want to try the experimental feature ?

Comment (abchoo) [reply config/configmap/inferenceservice.yaml:34]: I think that would be a good idea. We can also update the docs to include that as part of the how to start.

**Pull request merged**